src/unstated.js

```diff
@@ -139,3 +139,31 @@ export function Provider(props: ProviderProps) {
     </StateContext.Consumer>
   );
 }
+
+type ContainersMap = {
+  [string]: *
+};
+
+export const withSubscription = (containersMap: ContainersMap) => (
+  WrappedComponent: ComponentType<*>
+) => (props: *) => {
+  const containerKeys = Object.keys(containersMap);
+  const containers = containerKeys.map(
+    containerKey => containersMap[containerKey]
+  );
+
+  return (
+    <Subscribe to={containers}>
+      {(...containerInstances) => {
+        const containerProps = containerInstances.reduce(
+          (accumulator, instance, index) =>
+            Object.assign({}, accumulator, {
+              [containerKeys[index]]: instance
+            }),
+          {}
+        );
+        return <WrappedComponent {...props} {...containerProps} />;
```

Comment:
IMHO reversed order would be more appropriate - if one pass a prop explicitly that clashes with containerProps the former should take precedence.

Providing clashing names is probably not something that anyone would like to do and might be source of the confusion, so to mitigate this problem you could add a **development** warning just before `return` that would warn against such clashes.